### ISSUE TITLE
Visual Particle Shader overrides Transform in Process

### ISSUE THREAD

**Issue by kpfries** [godotengine/godot#56358]
### Godot version

4.0.dev.custom_build [9f058674a]

### System information

Windows 10, Ryzen 5 2600, GTX 1060, Vulkan

### Issue description

The visual shader editor for particles resets the particle transform to the generated matrix in the process function. I'd expect it to work like the start function where it generates a transform and multiplies it by the original transform. 

This is code generated for rotation and scale respectively in the start function
`TRANSFORM *= __build_rotation_mat4(vec3(0, 1, 0), n_out4p0);
TRANSFORM *= mat4(vec4(n_out5p0, 0, 0, 0), vec4(0, n_out5p0, 0, 0), vec4(0, 0, n_out5p0, 0), vec4(0, 0, 0, 1));`

This is a generated scale operation in the process function
`TRANSFORM = mat4(vec4(n_out2p0, 0, 0, 0), vec4(0, n_out2p0, 0, 0), vec4(0, 0, n_out2p0, 0), vec4(0, 0, 0, 1));`

### Steps to reproduce

If you plug any values into the scale or rotation inputs in the visual shader editor's process function in particle mode the transform will be reset. 

### Minimal reproduction project

[reproduction_project.zip](https://github.com/godotengine/godot/files/7794794/reproduction_project.zip)

**Comment by Calinou** [comment]
cc @Chaosus 

Can you reproduce this in 3.5?

**Comment by kpfries** [comment]
No, unless I'm doing something wrong the 3.4 particle visual shader system is pretty different. It doesn't have a separate process function and only takes a single input for the transformation matrix instead of separate float inputs for rotation and scale. 
![image](https://user-images.githubusercontent.com/5532428/147793076-1e55781c-1414-4292-9ece-679c19973ea8.png)

**Comment by Chaosus** [comment]
I've expected that too but changed it after this comment: https://github.com/godotengine/godot/pull/42248#issuecomment-817284178. cc @QbieShay - I'm not sure if there is a point in not multiplying this matrix (only if the connection to these ports exist of course).

**Comment by QbieShay** [comment]
When i wrote that comment i made a mistake: I was specifically talking about the basis of the particle rather than the whole transform. My bad, sorry!

**Comment by QbieShay** [comment]
Bumping this to 4.1 . While i understand this is a frustrating issue to deal with, there's no easy fix other than the linked PR, which is a bit too involved to be merged at RC stage.
